### ISSUE TITLE
Website optimization

### ISSUE THREAD

**Issue by Cepheid-UK** [w3champions/w3champions-ui#315]
Hello this is a call to arms. This thread will be for discussing potential improvements to the website to prevent the 500 errors we've been having. A good portion of these are due to the server overloading, and for sure there are things we can do to improve that.

**Comment by Cepheid-UK** [comment]
- When loading "/Ranking" page, requests are made for all ongoing matches, this is usually 50+ games during peak times, and is usually the heaviest request on this page. Currently this is necessary for the purpose of loading the games with streamers. Instead of querying every ongoing match, perhaps we can store who is in a game and streaming in a separate collection and update that with matchCreatedEvents and matchFinishedEvents, and query that (much smaller) collection instead of the entire ongoing matches.

**Comment by Cepheid-UK** [comment]
- This request for ongoing matches is happening EVERY 10 SECONDS for EVERY OPEN TAB. This seems a tad excessive to say the least. Maybe once every 30 seconds, and some way of preventing multiple tabs taking down the server?
- 
![ongoingmatchrequests](https://user-images.githubusercontent.com/40397054/102725212-f12f2780-430c-11eb-8565-2bdbe30167ce.PNG)

**Comment by Cepheid-UK** [comment]
- On the overall statistics tab, we're requesting ALL of the statistics, even the ones for the other tabs. We should really only request as we use them. "map-race-wins" is a particular offended @ 801 Kb and is only even used on the tab "Winrates" as far as I can tell.

**Comment by SergeyKhval** [comment]
@Cepheid-UK I pushed a fix here https://github.com/w3champions/w3champions-ui/commit/63652a7930c2022e3e8cb9d7d7314a237d2cb2b9

1. There was a problem with overall routing. Any route change would result in full component tree rerender. It means that all lifecycle hooks would trigger. On statistics page it led to refetching all data on any tab switch
2. I separated requests on stats pages. So now heave requests will run only when needed
3. I added caching for some views. To be precise for tabs on profile and stats pages. So now once data is loaded for those pages it will live forever until page refresh. It means switching between tabs will be almost instantaneous once data is loaded
